ctru-sys/build.rs

```diff
@@ -14,4 +16,69 @@ fn main() {
             _ => "ctru",
         }
     );
+
+    match check_libctru_version() {
+        Ok((maj, min, patch)) => {
+            eprintln!("using libctru version {maj}.{min}.{patch}");
+
+            // These are accessible by the crate during build with `env!()`.
+            // We might consider exporting some public constants or something.
+            println!("cargo:rustc-env=LIBCTRU_VERSION={maj}.{min}.{patch}");
+            println!("cargo:rustc-env=LIBCTRU_MAJOR={maj}");
+            println!("cargo:rustc-env=LIBCTRU_MINOR={min}");
+            println!("cargo:rustc-env=LIBCTRU_PATCH={patch}");
+
+            // It would be nice if we could rerun-if-changed on libctru itself,
+            // maybe we can write a file to OUT_DIR and check that file?
```

Comment:
I don't think that will work since we're the ones changing the file, so it won't change if libctru changes. Instead, we could do a `rerun-if-changed` on the `libctru` static library file. We'd just have to resolve that path.

Reply:
Good point, we already know the library name / search dirs we're linking against so it should be possible to figure out the filename of the actual library being used, which would change if the libs got updated. Ideally we could detect header changes too, but maybe just library is enough...

I suppose in theory we could also take the output of `[dkp-]pacman -Ql libctru` to list _all_ files owned by the package, but I'm not sure if that provides any significant advantage over just using the library. I guess it would include docstring changes ?